### ISSUE TITLE
Map Generation and Water Metrics

### ISSUE THREAD

**Issue by taylaward** [UQdeco2800/2022-studio-1#30]
# **Description**
This feature regards the implementation of the base map and environment that the player will discover, explore, and develop on. Vital to the overall game, the map provides the basis for the construction of environment features (cliffs, buildings, etc.), and further develops the sense of immersion by providing design elements that are suggestive of the overall ancient/ruins theme. This feature will focus on how the map is generated, how the user can interact with the map (e.g. building grid), and the design of the map at each phase of the game. The basis for the map is a collection of uniquely styled tiles that will be used to procedurally-generate a base layer. Each tile will represent a cell in a grid, on which the player can place any building, so long as there are enough cells in the grid to build that object.

## SubFeature/component (optional)
Generation:
The map will be generated at the beginning of each new game as a collection of tiles generated by an algorithm that groups certain elements together (paths, grass, sand, dirt, etc.). This map will become slowly discovered as the water is pushed further and further back. The map will only be generated once per game.
Interaction:
Each tile/cell in the grid will have an 'occupied' variable, which will determine whether or not the tile can be built upon.

# **Dependencies**

List any features on which this ticket relies. If there is already an existing ticket, reference it using #xx. If a feature needs to be implemented and the issue does not already exist... make a new ticket and leave it unassigned for someone else to take on.

e.g. Random game mode relies on random map seed generation(#xx).

# **Milestones**
List of sub-features you intend to have done and in Master by the end of the sprint.
- [ ] Goal 1 (Aug. 20)

- Initialize the game map 
-- Generating Ground Level
-- 3 Dimensions
--- Width, Length and Height
- Design pixel art for single block of ground
-- Different angle
-- differ looks
-- colours(dimmer when users on different level)

- [ ] Goal 2 (Aug. 23)
- Restrict adding Map Height level
-- Only once a structure has been built on current height, then another height can be added to the map
- Design pixel art for single block of ground - Day time / Night time
-- Ground (sand and soil)
-- Water
-- Grass    
-- sky

- [ ] Goal 3 (Aug. 26)
- Implement water rising and how it affects height levels
-- Rules, water can only go up or down one level, can’t skip a level etc.
- Start on background design

- [ ] Goal 4 (Aug. 28)
- Map design
-- Standard map - adding forests, water resources etc on the map.
-- Making a standard map for the first sprint.

**Completion Deadline:** Aug. 30

# **Extras (optional)**
- [ ] Minimap
- [ ] Map seeds

# **Documentation**

# **Members**
- Emily-Kate Williams
- Jae Won Yun (@alexyun0429) (JaeWon Yun)
- Pragun Ramesha (@Xodacan) (Pragun Ramesha)
- Sam Behm
- Tayla Ward (@taylaward) (Tayla Ward)

**Comment by luke4641942** [comment]
Just a quick revision about what we (team 5) discussed with you guys in the studio on Wednesday. We spoke about how you guys are planning to put collidable landforms in/on some of your blocks where buildings cant be placed. If it can be implemented with main character, we spoke about the main character being able to mine (turn the block to grass instead and collect the resources) what's on the block so that buildings could then be placed there and collect resources at the same time.
